microcosm_pubsub/metrics.py

```diff
@@ -51,3 +54,53 @@ def __call__(self, result):
             result.elapsed_time,
             tags=tags,
         )
+
+
+@defaults(
+    enabled=typed(boolean, default_value=True)
+)
+class PubSubSendBatchMetrics:
+
+    def __init__(self, graph):
+        self.metrics = self.get_metrics(graph)
+        self.enabled = bool(
+            self.metrics
+            and self.metrics.host != "localhost"
+            and graph.config.pubsub_send_metrics.enabled
+        )
+
+    def get_metrics(self, graph) -> Optional[DogStatsd]:
+        """
+        Fetch the metrics client from the graph.
+
+        Metrics will be disabled if the not configured.
+
+        """
+        try:
+            return graph.metrics
+        except NotBoundError:
+            return None
+
+    def __call__(self, elapsed_time, message_count: int):
+        """
+        Send metrics if enabled.
+
+        """
+        if not self.enabled:
+            return
+
+        tags = [
+            "source:microcosm-pubsub",
+        ]
+
+        self.metrics.histogram(
+            "message_batch",
+            elapsed_time,
+            tags=tags,
+        )
+
+        self.metrics.histogram(
+            "message_batch_count",
```

Comment:
Ends up being the approximate count per-batch of events that were actually handled (e.g. checks proportion of events out of the default 5/10 that were relevant).